.github/workflows/e2e-instances.yml

```
@@ -0,0 +1,302 @@
+name: Testing
+
+on: [pull_request]
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  prepare_variables:
+    runs-on: ubuntu-22.04
+    outputs:
+      POOL_NAME: ${{ steps.generate.outputs.POOL_NAME }}
+      POOL_INSTANCE: ${{ steps.generate.outputs.POOL_INSTANCE }}
+      BASE_URL: ${{ steps.generate.outputs.BASE_URL }}
+      API_URI: ${{ steps.generate.outputs.API_URI }}
+      BACKUP_ID: ${{ steps.backup.outputs.BACKUP_ID }}
+      BACKUP_VER: ${{ steps.backup.outputs.BACKUP_VER }}
+      BACKUP_NAME: ${{ steps.backup.outputs.BACKUP_NAME }}
+
+    steps:
+      - uses: actions/checkout@v4
```

Can do a partial checkout as only `.github/actions` is being used:

```suggestion
      - uses: actions/checkout@v4
        with:
          sparse-checkout: ./.github/actions
```